src/requests/utils.ts

```diff
@@ -0,0 +1,45 @@
+import * as querystring from 'querystring'
+
+export const getNormalizedEncodedURI = (uri: string) => {
+  const [pathname, search] = uri.split('?')
+  const escapedSearch = search ? querystring.escape(search) : ''
+
+  return encodeURI(escapedSearch ? `${pathname}?${escapedSearch}` : pathname)
+}
+
+export const sortHeaderKeys = (keyA: string, keyB: string) => (keyA > keyB ? 1 : -1)
+
+const normalizeHeaderKey = (key: string) => key.toLowerCase().trim()
+const normalizeHeaderValue = (value: string) => value.trim()
+export const normalizeHeaders = (headers: Record<string, string>) =>
+  map(headers, ([key, value]) => [normalizeHeaderKey(key), normalizeHeaderValue(value)])
+
+export const pickHeaders = (headers?: Record<string, string>, keys?: string[]) => {
+  if (!headers) {
```

Comment:
As far as I see there's no code path that ends up with `headers` or `keys` not defined, so this function might be unnecessary or ends up being just an alias for `filter()`